frameworks/non-keyed/miso/src/Main.hs

```diff
@@ -0,0 +1,306 @@
+
+updateModel (ChangeModel newModel) _ = noEff newModel
+
+updateModel (Create n) model@Model{lastId=lastIdx} =
+  model <# do
+    newRows <- generateRows n lastIdx
+    pure $ ChangeModel model  { rows = newRows
+                              , lastId = lastIdx + n
+                              }
+
+updateModel (Append n) model@Model{rows=existingRows, lastId=lastIdx} =
+  model <# do
+    newRows <- generateRows n (lastId model)
+    pure $ ChangeModel model  { rows=existingRows V.++ newRows
+                              , lastId=lastIdx + n
+                              }
+
+updateModel Clear model = noEff model{ rows= V.empty }
+
+updateModel (Update n) model =
+  noEff model{ rows = updatedRows }
+  where
+      updatedRows = V.imap updateR (rows model)
+      updateR i row = if mod i 10 == 0
+                          then row{ rowTitle = rowTitle row <> " !!!" }
+                          else row
+
+updateModel Swap model =
+  noEff newModel
+  where
+    currentRows = rows model
+    from = V.indexed
+    newModel = if V.length currentRows > 998
+                  then model { rows = swappedRows }
+                  else model
+    swappedRows = V.modify (\v -> MV.swap v 1 998) currentRows
+
+updateModel (Select idx) model = noEff model{selectedId=Just idx}
+
+updateModel (Remove idx) model@Model{rows=currentRows} =
+  noEff model { rows = firstPart V.++ V.drop 1 remainingPart }
+  where
+    (firstPart, remainingPart) = V.splitAt idx currentRows
+
+updateModel NoOp model = noEff model
+
+generateRows :: Int -> Int -> IO (V.Vector Row)
+generateRows n lastIdx = V.generateM n $ \x -> do
+  adjIdx <- randomRIO (0, V.length adjectives - 1)
+  colorIdx <- randomRIO (0, V.length colours - 1)
+  nounIdx <- randomRIO (0, V.length nouns - 1)
+  pure Row
+          { rowIdx=lastIdx + x
+          , rowTitle= (adjectives V.! adjIdx)
+                      <> S.pack " "
+                      <> (colours V.! colorIdx)
+                      <> S.pack " "
+                      <> (nouns V.! nounIdx)
+          }
+
+viewModel :: Model -> View Action
+viewModel m = div_ [id_ "main"]
+  [ div_
+      [class_ "container"]
+      [ viewJumbotron
+      , viewTable m
+      , span_ [class_ "preloadicon glyphicon glyphicon-remove", textProp "aria-hidden" "true"] []
+      ]
+  ]
+
+viewTable :: Model -> View Action
+viewTable m@Model{selectedId=idx} =
+  table_
+    [class_ "table table-hover table-striped test-data"]
+    [
+      tbody_
+        [id_ "tbody"]
+        (V.toList $ V.imap viewRow (rows m))
+    ]
+  where
+    viewRow i r@Row{rowIdx=rId} =
+      tr_
+        (conditionalDanger i)
+        [ td_
+            [ class_ "col-md-1" ]
+            [ text (S.ms rId) ]
+        , td_
+            [ class_ "col-md-4" ]
+            [ a_ [class_ "lbl", onClick (Select i)] [text (rowTitle r)]
+            ]
+        , td_
+            [ class_ "col-md-1" ]
+            [ a_
+              [ class_ "remove" ]
+              [ span_
```

Comment:
aria-hidden="true" is missing here, isn't it?

Reply:
And similarly in the keyed version. (Leaving out the aria-hidden attribute in the table might be taken as a shortcut,  I currently do not test it for the six buttons)

Reply:
You fixed it ! Thanks ;) @krausest